lib/shopify-cli/process_supervision.rb

```diff
@@ -104,10 +110,12 @@ def running?(identifier)
       end
     end
 
-    def initialize(identifier, pid:, time: Time.now.strftime('%s')) # :nodoc:
+    def initialize(identifier, pid: nil, time: Time.now.strftime('%s')) # :nodoc:
```

Comment:
why can pid be nil now?

Reply:
The pid can be nil because we need this object before we call `spawn`, which returns the pid that gets set to it (via the new accessor above).